src/vmm/src/lib.rs

```diff
@@ -1546,14 +1546,7 @@ impl Vmm {
                 let new_size = File::open(&drive_config.path_on_host)
                     .and_then(|mut f| f.seek(SeekFrom::End(0)))
                     .map_err(|_| DriveError::BlockDeviceUpdateFailed)?;
-                if new_size % virtio::block::SECTOR_SIZE != 0 {
-                    warn!(
-                        "Disk size {} is not a multiple of sector size {}; \
-                         the remainder will not be visible to the guest.",
-                        new_size,
-                        virtio::block::SECTOR_SIZE
-                    );
-                }
+                virtio::block::check_disk_size(new_size);
```

Comment:
I d place the `check_disk_size` function inside the `build_config_space` function since these two are tightly coupled. Every time you look at size you will also end up modifying the config, right? In  this scenario the vmm calls into the device manager which ultimately calls into the device for rebuilding config space.

Reply:
I moved the warning from the `check_disk_size` function inside `build_config_space`.